src/b3d/bucket_utils/b3d_pull.py

```diff
@@ -54,25 +54,15 @@ def download_all(overwrite):
     )
     print("===============================================")
 
-    if overwrite:
-        download_cmd = [
-            "gcloud",
-            "storage",
-            "cp",
-            "--recursive",
-            GCLOUD_BUCKET_NAME + "/*",
-            LOCAL_BUCKET_PATH,
-        ]
-    else:
-        download_cmd = [
-            "gcloud",
-            "storage",
-            "cp",
-            "-n",
-            "--recursive",
-            GCLOUD_BUCKET_NAME + "/*",
+    flags = [] if overwrite else ["-u"]
```

Comment:
this `-u` flag will cause the command to NOT update a local file if its timestamp is newer. Without this flag, the remote version, if different, will always win.